native/src/main/java/io/ballerina/stdlib/xmldata/MapFromXml.java

```diff
@@ -97,8 +98,9 @@ public static Object fromXml(BXml xml, BTypedesc type) {
                     for (Map.Entry<BString, Object> entry : resultMap.entrySet()) {
                         BTable tableValue = ValueCreator.createTableValue(tableType);
                         Type tableValueType = ((TableType) valueType).getConstrainedType();
-                        if (tableValueType.getTag() == TypeTags.RECORD_TYPE_TAG) {
-                            tableValue.put(CloneWithType.convert(tableValueType, entry.getValue()));
+                        if (tableValueType.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG) {
+                            tableValue.put(CloneWithType.convert(TypeUtils.getReferredType(tableValueType),
+                                    entry.getValue()));
```

Comment:
Here you have assumed that only records are of reference type.
What would happen if the reference type is a base type
```
int i = 5;
type IntValue i;
```
Wouldnt it be better to resolve the referred type before any log?

```suggestion
                         Type tableValueType = Type tableValueType = ((TableType) valueType).getConstrainedType());
                          if (tableValueType.getTag() == TypeTags.RECORD_TYPE_TAG) {
                            tableValue.put(CloneWithType.convert(tableValueType,
                                    entry.getValue()));
```

Reply:
+1
And we don't need to specifically check for `TypeTags.TYPE_REFERENCED_TYPE_TAG` as the `TypeUtils.getReferredType()` handles it internally. 
```suggestion
                        Type tableValueType = TypeUtils.getReferredType(((TableType) valueType).getConstrainedType());
                           if (tableValueType.getTag() == TypeTags.RECORD_TYPE_TAG) {
                            tableValue.put(CloneWithType.convert(tableValueType,
                                    entry.getValue()));
```